R/update-news.R

```diff
@@ -125,3 +136,21 @@ edit_cran_comments <- function() {
   local_options(usethis.quiet = TRUE)
   edit_file("cran-comments.md")
 }
+
+is_merge_commit <- function(message) {
+  grepl("^Merge pull request #[0-9]* from", message)
+}
+
+harvest_pr_title <- function(message) {
+  pr_number <- regmatches(message, regexpr("#[0-9]*", message))
+  pr_number <- sub("#", "", pr_number)
+  slug <- github_slug()
+  pr_info <- try(gh::gh(glue("GET /repos/{slug}/pulls/{pr_number}")), silent = TRUE)
+
+  if (inherits(pr_info, "try-error")) {
+    cli::cli_alert_warning(sprintf("Could not get title for PR #%s", pr_number))
```

Comment:
Can we return a default title here that we can watch out for later? My vision of an offline operation mode:

- A single warning appears
- All PRs without a message of their own in the merge commit get a default message that also contains the PR number
- We provide an API that searches `NEWS.md` for those default messages and substitutes with the real thing

We need to think about when to expand conventional-commit messages. I wouldn't mind if this happened only during `pre_relesae()`, are there constraints that would prohibit this?

Loose thought: if a mix of cc and non-cc messages is detected, we can list the non-cc messages under their own header "uncategorized"?